kyuubi-server/src/main/scala/org/apache/kyuubi/server/KyuubiTHttpFrontendService.scala

```diff
@@ -0,0 +1,285 @@
+    conf.get(FRONTEND_THRIFT_HTTP_BIND_HOST)
+  override protected lazy val portNum: Int = conf.get(FRONTEND_THRIFT_HTTP_BIND_PORT)
+  override protected lazy val actualPort: Int = portNum
+  override protected lazy val serverSocket: ServerSocket = null
+
+  private var server: Option[Server] = None
+  private val APPLICATION_THRIFT = "application/x-thrift"
+
+  /**
+   * Configure Jetty to serve http requests. Example of a client connection URL:
+   * http://localhost:10000/servlets/thrifths2/ A gateway may cause actual target
+   * URL to differ, e.g. http://gateway:port/hive2/servlets/thrifths2/.
+   *
+   * @param conf the configuration of the service
+   */
+  override def initialize(conf: KyuubiConf): Unit = synchronized {
+    this.conf = conf
+    try {
+      // Server thread pool
+      // Start with minWorkerThreads, expand till maxWorkerThreads and reject
+      // subsequent requests
+      val minThreads = conf.get(FRONTEND_THRIFT_MIN_WORKER_THREADS)
+      val maxThreads = conf.get(FRONTEND_THRIFT_MAX_WORKER_THREADS)
+      val keepAliveTime = conf.get(FRONTEND_THRIFT_WORKER_KEEPALIVE_TIME)
+      val executor = new ThreadPoolExecutor(
+        minThreads,
+        maxThreads,
+        keepAliveTime,
+        TimeUnit.MILLISECONDS,
+        new SynchronousQueue[Runnable](),
+        new NamedThreadFactory(getName + "HttpHandler-Pool", false))
+      val threadPool = new ExecutorThreadPool(executor)
+
+      // HTTP Server
+      server = Some(new Server(threadPool))
+
+      val httpConf = new HttpConfiguration
+      // Configure header size
+      val requestHeaderSize = conf.get(FRONTEND_THRIFT_HTTP_REQUEST_HEADER_SIZE)
+      val responseHeaderSize = conf.get(FRONTEND_THRIFT_HTTP_RESPONSE_HEADER_SIZE)
+      httpConf.setRequestHeaderSize(requestHeaderSize)
+      httpConf.setResponseHeaderSize(responseHeaderSize)
+      val connectionFactory = new HttpConnectionFactory(httpConf)
+
+      val useSsl = conf.get(FRONTEND_THRIFT_HTTP_USE_SSL)
+      val schemeName = if (useSsl) "https" else "http"
+
+      // Change connector if SSL is used
+      val connector =
+        if (useSsl) {
+          val keyStorePath = conf.get(FRONTEND_THRIFT_HTTP_SSL_KEYSTORE_PATH)
+
+          if (keyStorePath.isEmpty) {
+            throw new IllegalArgumentException(FRONTEND_THRIFT_HTTP_SSL_KEYSTORE_PATH.key +
+              " Not configured for SSL connection")
+          }
+
+          val keyStorePassword = conf.get(FRONTEND_THRIFT_HTTP_SSL_KEYSTORE_PASSWORD)
+          if (keyStorePassword.isEmpty) {
+            throw new IllegalArgumentException(FRONTEND_THRIFT_HTTP_SSL_KEYSTORE_PASSWORD.key +
+              " Not configured for SSL connection")
+          }
+
+          val sslContextFactory = new SslContextFactory.Server
+          val excludedProtocols = conf.get(FRONTEND_THRIFT_HTTP_SSL_PROTOCOL_BLACKLIST).split(",")
+          info("HTTP Server SSL: adding excluded protocols: " +
```

Comment:
More readable log information? Kyuubi has owned an HTTP server to serve restful API. So let's make the `HTTP server` clear?